lib/hanami/application/routing/resolver.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Hanami
+  class Application
+    module Routing
+      # Hanami application router endpoint resolver
+      #
+      # @since 2.0.0
+      class Resolver
+        require "hanami/application/routing/resolver/trie"
+
+        # @api private
+        class NotCallableEndpointError < StandardError
+          def initialize(endpoint)
+            super("#{endpoint.inspect} isn't compatible with Rack. Please make sure it implements #call.")
+          end
+        end
+
+        attr_reader :container
+        attr_reader :base_namespace
+        attr_reader :inflector
+        attr_reader :slices
+
+        def initialize(container:, namespace:, inflector:, slices: Trie.new)
+          @container = container
+          @base_namespace = namespace
+          @inflector = inflector
+          @slices = slices
+        end
+
+        def with_container(new_container)
+          self.class.new(
+            container: new_container,
+            namespace: base_namespace,
+          )
+        end
```

Comment:
Seems like this entire method isn't needed anymore? It existed previously because of the way we did the nesting of the router objects to achieve the old way of slice mounting. Can remove?

Reply:
Yes, removed by cc3b063